app/uk/gov/hmrc/agentsubscription/model/InternalId.scala

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.agentsubscription.model
+
+import play.api.libs.json.Format
+import play.api.libs.functional.syntax._
+
+/**
+ * An internal id associated with a Government Gateway account.
+ *
+ * @param id
+ */
+final case class InternalId(id: String)
+
+object InternalId {
+  implicit val format: Format[InternalId] = implicitly[Format[String]].inmap(InternalId(_), _.id)
```

Comment:
wouldnt `implicit val format: Format[InternalId] = Json.format[InternalId]` work ?

Reply:
We could use that, but the downside is that we'd get extra redundant structure in our JSON ("id -> "xxx").  This method basically translates directly to/from a String using the invariant map, so we just get "internalId" -> "xxx"

Reply:
ok, no worries, i remember using SimpleObjectReads or something, but thats fine